samples/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgismaps/sample/editfeatureswithfeaturelinkedannotation/components/EditFeaturesWithFeatureLinkedAnnotationViewModel.kt

```diff
@@ -0,0 +1,330 @@
+        val parcelAnnotationTable = geodatabase.annotationTables.find { it.tableName == parcelAnnoTableName }
+        // Create annotation layers from the tables
+        val addressAnnotationLayer = addressAnnotationTable?.let { AnnotationLayer(it) }
+        val parcelAnnotationLayer = parcelAnnotationTable?.let { AnnotationLayer(it) }
+        // Add layers to map
+        arcGISMap.operationalLayers.clear()
+        listOfNotNull(addressFeatureLayer, parcelFeatureLayer, addressAnnotationLayer, parcelAnnotationLayer).forEach {
+            arcGISMap.operationalLayers.add(it)
+        }
+    }
+
+    /**
+     * Handle single tap on the map.
+     * If a feature is already selected, treat as move confirmation.
+     * Otherwise, identify the feature at the tapped location.
+     */
+    fun onMapSingleTap(screenCoordinate: ScreenCoordinate, mapPoint: Point) {
+        // If a feature is already selected, treat as move confirmation
+        if (selectedFeature != null) {
+            // Only allow move for point or straight polyline
+            val feature = selectedFeature ?: return
+            if (feature.geometry is Point || isStraightPolyline(feature.geometry)) {
+                onMoveFeatureConfirmed(feature, mapPoint)
+            } else {
+                clearSelection()
+                _instruction.value = Instruction.SelectFeature
+            }
+            return
+        }
+        // Otherwise, identify feature at tap
+        viewModelScope.launch {
+            val results = mapViewProxy.identifyLayers(
+                screenCoordinate = screenCoordinate, tolerance = 10.dp, returnPopupsOnly = false, maximumResults = 1
+            ).getOrElse {
+                messageDialogVM.showMessageDialog(it)
+                return@launch
+            }
+            selectFirstFeature(results)
+        }
+    }
+
+    /**
+     * Select the first feature from the identify results.
+     * If it's a point, show the edit address dialog.
+     * If it's a straight polyline, allow moving it.
+     * Otherwise, clear selection and show instruction to select a feature.
+     */
+    private fun selectFirstFeature(results: List<IdentifyLayerResult>) {
+        clearSelection()
+        // Prefer feature layers only
+        val featureLayerResult = results.firstOrNull {
+            it.layerContent is FeatureLayer && it.geoElements.isNotEmpty()
+        } ?: return
+        val featureLayer = featureLayerResult.layerContent as? FeatureLayer ?: return
+        val feature = featureLayerResult.geoElements.firstOrNull() as? ArcGISFeature ?: return
+        featureLayer.selectFeature(feature)
+        selectedFeature = feature
+        when (feature.geometry) {
+            is Point -> {
+                showEditAddressDialog = true
+                // Prepare address editing
+                val adAddress = feature.attributes["AD_ADDRESS"] as Int
+                val stStrNam = feature.attributes["ST_STR_NAM"]?.toString() ?: ""
```

Comment:
Should we add these attribute keys with the other constants list on the top the viewmodel?